src/main/java/seedu/toluist/controller/AddTaskController.java

```diff
@@ -78,52 +79,68 @@ private CommandResult add(TodoList todoList, String description,
             LocalDateTime eventStartDateTime, LocalDateTime eventEndDateTime,
             LocalDateTime taskDeadline, String taskPriority, Set<Tag> tags,
             String recurringFrequency, LocalDateTime recurringUntilEndDate) {
-        if (!isValidTaskType(eventStartDateTime, eventEndDateTime, taskDeadline)) {
-            return new CommandResult(RESULT_MESSAGE_ERROR_DATE_INPUT);
-        }
-
-        Task task;
-        if (eventStartDateTime != null && eventEndDateTime != null) {
-            task = new Task(description, eventStartDateTime, eventEndDateTime);
-        } else if (taskDeadline != null) {
-            task = new Task(description, taskDeadline);
-        } else {
-            task = new Task(description);
+        if (!StringUtil.isPresent(description)) {
+            return new CommandResult(RESULT_MESSAGE_ERROR_EMPTY_DESCRIPTION);
         }
-        if (taskPriority != null) {
-            task.setTaskPriority(taskPriority);
-        }
-        if (StringUtil.isPresent(recurringFrequency)) {
-            if (recurringUntilEndDate == null) {
-                task.setRecurring(recurringFrequency);
+        try {
+            // validates that the dates input belongs to only one type of task, or exception is thrown
+            validateTaskDatesInput(eventStartDateTime, eventEndDateTime, taskDeadline);
+            Task task;
+            if (eventStartDateTime != null && eventEndDateTime != null) {
```

Comment:
They are not exact. Update can update either from/ or to/
Add must have both arguments.